tensorflow/contrib/layers/python/layers/layers.py

```diff
@@ -3117,7 +3117,7 @@ def maxout(inputs, num_units, axis=-1, scope=None):
       raise ValueError('number of features({}) is not '
                        'a multiple of num_units({})'.format(
                            num_channels, num_units))
-    shape[axis] = -1
+    shape[axis] = num_units
```

Comment:
Can we add a test case with `None` value for an axis?

Reply:
Like this? I am not sure.
```python
input_tensor =tf.placeholder(tf.float32, [None, 32, 32, 3], name="input_data")
with tf.variable_scope('conv_layer'):
    conv_out = tf.layers.conv2d(input_tensor, 10, (5,5), padding='same', name="conv")
    maxout_out = tf.contrib.layers.maxout(conv_out_1, 5, axis=3)
#... some test code.
self.assertEqual(maxout_out.get_shape().as_list()[3], 5, "shape in maxout function failed")
```

Reply:
```python
import numpy as np
import tensorflow as tf

# Suppose this function belong to some test class
def testMaxoutShape(self):
    data = np.random.randn(64, 32, 32, 3).astype(np.float32)
    input_tensor = tf.placeholder(tf.float32, [None, 32, 32, 3], name="input_data")
    conv_out = tf.layers.conv2d(input_tensor, 10, (5,5), padding='same', name="conv")
    maxout_out = tf.contrib.layers.maxout(conv_out, 5, axis=3)

    with tf.Session() as sess:
        sess.run(tf.global_variables_initializer())
        value = sess.run(maxout_out, feed_dict={input_tensor: data})
    self.assertEqual(value.shape[3], 5, "shape in maxout function failed")
```

Reply:
Would you like to make the change to the test file? https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/layers_test.py#L4172